tutorial/practicals/LFRic/building_code/4_psydata/README.md

```diff
@@ -0,0 +1,205 @@
+    CALL extract_psy_data%ProvideVariable("perturbation_post", perturbation)
+    CALL extract_psy_data%PostEnd
+
+Notice that the variable 'perturbation' is provided once before the kernel call
+with the name ``perturbation``, but then again after the invocation with
+the name ``perturbation_post``. This variable is an input- and output-variable,
+so its values are stored twice, just with a different name.
+
+
+## Step 5: Run the application
+After running the application using:
+
+    ./time_evolution
+
+you should get a new NetCDF file called
+``time_evolution_alg_mod_psy-invoke_propagate_perturbation:prop_perturbation_code:r0.nc``.
+This is just the concatenation of the two parameters to the ``PreStart`` function.
+The long and convoluted name is caused by PSyclone automatically creating a unique module 
+and local name. 
+
+# Step 6: Provide a more user friendly name
+You can provide a more user friendly name by providing a
+``region_name`` as an option to the transformation. The region name is a pair of strings,
+the first one being a module name, the second a region name. You can use them in any
+way you like. The options are provided as an additional dictionary to the apply function
+with "region" as key, and first a module name, then a local name. You could use:
+
+    {"region_name": ("time_evolution", "propagate")}
+
+Add this dictionary to your transformation script, and rebuild your application. You
+have to do a
+
+    make -f Makefile.extract clean
+
+since the Makefile does not have a dependency on your script (though it would be
+recommended to add this to the Makefile so that a change in your script automatically
+triggers running PSyclone again).
+
+You should now see that the psy-layer contains:
+
+    CALL extract_psy_data%PreStart("time_evolution", "propagate", 7, 2)
+
+I.e. it uses the names you have provided, and running the binary will now create
+a NetCDF file called ``time_evolution-propagate.nc``.
+
+## Step 7: Examining the output
+
+The NetCDF command ``ncdump`` can be used to check the content of the created file:
+
+    ncdump time_evolution-propagate.nc | less
+
+You will see that the dimension are defined for the fields used. Some variables
+are listed with the prefix ``post``, e.g. ``perturbation_post``. These are the
+values of the variable after the kernel call, so it allows the NetCDF file
+to store perturbation as input parameter (just ``perturbation``), and its
+values after the kernel (``perturbation_post``).
+
+At this stage no driver is created that can read in the file.
+
+
+## Step 8: Instrument more than one invoke
+
+The transformation script can be slightly changed to instrument all invokes
+in a file. While this might not be useful for kernel extraction, it is
+essential for e.g. parameter verification, where you typically want to
+check all invokes contained in a program. Use the template ``instrument_all.py``
+to apply the extraction transformation to all invokes. This script actually
```

Comment:
Handholding again.

Reply:
Same issue ... not sure what else to do. The template they should copy contains:
```
    # ------------------------------------------------------
    # TOOD: import the transformation and create an instance
    # ------------------------------------------------------
    # from ... import ...
    # my_transform = ...()
...
        # ------------------------------------------------------
        # TODO: Apply the transformation
        # ------------------------------------------------------
        ....apply(schedule, {"region_name": ("time_evolution", name)})
```
(and the options parameter have been explained before, and will be on the slides).

I don't really know how to hold hands ... they do have to take two steps themselves:)

Reply:
OK, I've mentioned the two steps in more detail.